### ISSUE TITLE
RealmSwift Does not build from source with Xcode 10.2 as Swift 3 support has been dropped.

### ISSUE THREAD

**Issue by james-rantmedia** [realm/realm-cocoa#6115]
## Goals
Build RealmCocoa (Realm and RealmSwift) from source with Xcode 10.2 (no longer in beta as of 26th March 2019)

## Expected Results
Expected Realm to build from source.

## Actual Results
The build fails, because Swift 3 support has been dropped from the Swift compiler included with Xcode 10.2.

```
/usr/bin/xcrun xcodebuild -workspace /Users/james/dev/MyApp/Carthage/Checkouts/realm-cocoa/Carthage/Realm.xcworkspace -scheme RealmSwift -configuration Release -derivedDataPath /Users/james/Library/Caches/org.carthage.CarthageKit/DerivedData/10.2_10E125/realm-cocoa/v3.13.1 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/89/zl8dhxjd64v9j1rrwhhpvbf80000gn/T/realm-cocoa SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/james/dev/MyApp/Carthage/Checkouts/realm-cocoa)User defaults from command line:
    IDEArchivePathOverride = /var/folders/89/zl8dhxjd64v9j1rrwhhpvbf80000gn/T/realm-cocoa
    IDEDerivedDataPathOverride = /Users/james/Library/Caches/org.carthage.CarthageKit/DerivedData/10.2_10E125/realm-cocoa/v3.13.1

Build settings from command line:
    CARTHAGE = YES
    CLANG_ENABLE_CODE_COVERAGE = NO
    CODE_SIGN_IDENTITY = 
    CODE_SIGNING_REQUIRED = NO
    GCC_INSTRUMENT_PROGRAM_FLOW_ARCS = NO
    ONLY_ACTIVE_ARCH = NO
    SDKROOT = iphoneos12.2
    SKIP_INSTALL = YES
    STRIP_INSTALLED_PRODUCT = NO

note: Using new build system
note: Planning build
note: Constructing build description
Build system information
error: SWIFT_VERSION '3.0' is unsupported, supported versions are: 4.0, 4.2, 5.0. (in target 'RealmSwift')

Build system information
error: SWIFT_VERSION '3.0' is unsupported, supported versions are: 4.0, 4.2, 5.0. (in target 'RealmSwift')
```

## Steps to Reproduce
Attempt to build Realm from source using the command line tools included with Xcode 10.2

## Version of Realm and Tooling
Realm framework version: 3.13.1
Xcode version: 10.2
iOS/OSX version: macOS 10.14.3, iOS 12.2

Dependency manager + version: Carthage 0.32.0

**Comment by nikolay-kapustin** [comment]
Swift 4.2 works fine, just try it.

**Comment by james-rantmedia** [comment]
I don't know what you mean.

My project is written in Swift 4.2. Realm appears to be using Swift 3 (it specifies Swift 3 in the swift_version build setting in its Xcode Project).
I updated to Xcode 10.2, opened my project and was told that Realm needed to be recompiled (cannot import module compiled using Swift 4.2 into Swift 5, etc), so I updated the Carthage builds and the build fails saying swift 3 is no longer supported.

**Comment by samueleperricone** [comment]
Waiting for the official update, I just altered the Realm Xcode project from the Carthage/Checkout folder settings every single Xcode target to use Swift 4.2.

Then I run `carthage build realm-cocoa --platform iOS`.

its working.

**Comment by mjhatamy** [comment]
@samueleperricone No, it is not working as expected.  It compiles but it is buggy, because of some missing internal linking .
 Also, Simulator code compile breaks as well.
 Xcode 10.2 uses Swift 5 compiler and even by using Swift 4.2, you still need to compile it with swift compiler 5 to be able to use the framework.

Realm on Xcode 10.2, (Swift 5 compiler) , even by Swift 4.2 currently impossible. 
Yes, I have it working on a real Device, but Some parts still fails.

 The other issue is the Realm Static files which are also not aligned and optimized for Swift 5 compilation

**Comment by james-rantmedia** [comment]
Yes exactly. I saw that there was a pull request to update to Swift 5, but it mentioned that many of the tests failed. I don't believe the solution is as simple as changing the swift version in the Xcode Project, I believe that is more likely to create hidden problems when using Realm in that manner.

For now I have had to revert back to Xcode 10.1.

**Comment by samueleperricone** [comment]
@james-rantmedia @mjhatamy It is not a permanent solution, as I started by comment "Waiting for the official update".

So this is just a temporary workaround to continue the local development until the pull request to update to Swift 5 is approved and merged.

**Comment by mjhatamy** [comment]
@samueleperricone Correct, but still even local development behave unexpectedly. My Experience was corrupted Realm Files. As long as your development does not rely on Realm it is fine.

**Comment by PushpakN** [comment]
Waiting eagerly for a stable release.

**Comment by mjhatamy** [comment]
Realm 3.14 is up and works on Swift 5. I am going to do more deeper tests

**Comment by humblehacker** [comment]
3.14 doesn't build for me with the Swift 5 compiler.  It segfaults trying to build `LinkingObjects.swift`.

Here's the stack trace:

**Comment by aregler** [comment]
It builds for me (carthage from master) but when building the project i get an error for each class which inherits from _Object_.

So the exact error message is:
> Cannot find interface declaration for 'RealmSwiftObject', superclass of 'WhateverRealmObject'

Does anybody else have this kind of problem?

**Comment by mjhatamy** [comment]
Do not compile current master. use tag 3.14.0

**Comment by kaiserabliz** [comment]
I am having the same issue as before, header 'RealmSwift-Swift.h' not found, with this new version.

**Comment by aregler** [comment]
> Do not compile current master. use tag 3.14.0

Thank you for the suggestion, unfortunately the error still remains.
I guess the difference is, that you are using Realm directly in your MainTarget where i use it in a framework. 

Reminds me somehow of: https://github.com/realm/realm-cocoa/issues/3073

**Comment by tgoyne** [comment]
A complete project which demonstrates issues with Xcode 10.2 and Realm 3.14 would be very helpful. I've been unable to reproduce any problems myself.

**Comment by aregler** [comment]
Hey @tgoyne,

here is a quick example app. 

Steps to reproduce:

- `carthage update --platform ios`
- build the test app

When you set "Install Objective-C Compability Header" to "No" you are able to build the app, but migration to Swift 5 fails.

Thank you for your effort good Sir!

[TestApp.zip](https://github.com/realm/realm-cocoa/files/3021760/TestApp.zip)

**Comment by james-rantmedia** [comment]
@tgoyne This issue was raised against 3.13.1, not 3.14.

If 3.14 solves the problems with Xcode 10.2 that's great, but it would be better for you to say that the issue has been resolved rather than asking for proof that there is any issue with a release that was created 3 days after this issue was raised.

**Comment by ricardohg** [comment]
@aregler having your exact same issue, setting "Compatibility Header" to "No" on the framework does the trick, but that doesn't seems right to me

**Comment by kaiserabliz** [comment]
I was able to build and run on simulator by disabling the Thread Sanitizer. It is actually mentioned on Swift 5 Release Notes on this site: [https://developer.apple.com/documentation/xcode_release_notes/xcode_10_2_release_notes/swift_5_release_notes_for_xcode_10_2?preferredLanguage=occ](url)

> - Known Issues
> The Swift compiler may crash during a build when the Thread Sanitizer is enabled. (48719789)
> Workaround: Disable Thread Sanitizer in the Scheme Editor’s Diagnostics tab.

Hope this helps.

**Comment by james-rantmedia** [comment]
The original problem this issue was raised for has been resolved since Realm 3.14 adds an Xcode 10.2 compatible binary for Carthage to download.

I would recommend that anyone experiencing the other issues being discussed in this thread open new issues to properly capture them.
